lib/decompress/zstd_decompress_block.c

```diff
@@ -280,12 +282,13 @@ size_t ZSTD_decodeLiteralsBlock(ZSTD_DCtx* dctx,
                     break;
                 case 1:
                     lhSize = 2;
+                    RETURN_ERROR_IF(srcSize<3, corruption_detected, "srcSize >= MIN_CBLOCK_SIZE == 2; here we need lhSize+1 = 3");
```

Comment:
This is correct. We have `+1` here because of the read of `istart[lhSize]` on lines 300 and 305. But in `set_basic` (above) we don't have the `+1`.

Reply:
Thanks!